autoparallel/graph_pp_runner.py

```diff
@@ -400,6 +454,19 @@ def __init__(
         schedule: _PipelineScheduleRuntime,
     ):
         self.schedule = schedule
+        if not schedule._backward_requires_autograd:
```

Comment:
can you say why we need this assertion?  what would happen if we had one graph stage and one non-graph stage?  (not that we want this, but, shouldn't it work fine?

Reply:
We are running the graph based pp runner in no_grad mode, so the no graph stage will not record any autograd graph for non-graph stages, hence we make this check.